cmd/hope/deploy.go

```diff
@@ -135,30 +138,60 @@ var deployCmd = &cobra.Command{
 					return err
 				}
 			case ResourceTypeDockerBuild:
-				// Strip the actual tag off the repo so that it defaults to the
-				//   latest.
-				tagSeparator := strings.LastIndex(resource.Build.Tag, ":")
-				pullImage := resource.Build.Tag
-				if tagSeparator != -1 {
-					pullImage = pullImage[:tagSeparator]
+				isCacheCommand := len(resource.Build.Source) != 0
```

Comment:
All this validation _should_ go somewhere more sensible. Basically only the deploy phase does any of this validation, and that's no good.